connector/spanmetricsconnector/factory.go

```diff
@@ -13,10 +13,28 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/connector"
 	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/featuregate"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/spanmetricsconnector/internal/metadata"
 )
 
+const (
+	DefaultNamespace               = "traces.span.metrics"
+	legacyMetricNamesFeatureGateID = "connector.spanmetrics.legacyMetricNames"
+)
+
+var legacyMetricNamesFeatureGate *featuregate.Gate
+
+func init() {
+	// TODO: Remove this feature gate when the legacy metric names are removed.
+	legacyMetricNamesFeatureGate = featuregate.GlobalRegistry().MustRegister(
+		legacyMetricNamesFeatureGateID,
+		featuregate.StageBeta,
```

Comment:
If you want This PR to be a breaking change, we need to do:
```suggestion
		featuregate.StageAlpha, // Alpha because we want it disabled by default.
```

Reply:
Should it be `StageBeta` or `StageAlpha`? I followed what you told me here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34485#discussion_r1714553911

I don't have a strong opinion.